src/NzbDrone.Core/Download/DownloadClientProvider.cs

```diff
@@ -18,17 +19,23 @@ public class DownloadClientProvider : IProvideDownloadClient
         private readonly Logger _logger;
         private readonly IDownloadClientFactory _downloadClientFactory;
         private readonly IDownloadClientStatusService _downloadClientStatusService;
+        private readonly IIndexerFactory _indexerFactory;
         private readonly ICached<int> _lastUsedDownloadClient;
 
-        public DownloadClientProvider(IDownloadClientStatusService downloadClientStatusService, IDownloadClientFactory downloadClientFactory, ICacheManager cacheManager, Logger logger)
+        public DownloadClientProvider(IDownloadClientStatusService downloadClientStatusService,
+                                      IDownloadClientFactory downloadClientFactory,
+                                      IIndexerFactory indexerFactory,
+                                      ICacheManager cacheManager,
+                                      Logger logger)
         {
             _logger = logger;
             _downloadClientFactory = downloadClientFactory;
             _downloadClientStatusService = downloadClientStatusService;
+            _indexerFactory = indexerFactory;
             _lastUsedDownloadClient = cacheManager.GetCache<int>(GetType(), "lastDownloadClientId");
         }
 
-        public IDownloadClient GetDownloadClient(DownloadProtocol downloadProtocol)
+        public IDownloadClient GetDownloadClient(DownloadProtocol downloadProtocol, int indexerId = 0)
```

Comment:
Or maybe we could store the indexer on the remote episode and use that directly instead of looking it up again? Forr release push we get the indexer and then store the name/ID and same for when it's set in `IndexerBase`.

Could repurpose `ReleaseInfo.Indexer` from a string to the indexer Definition, then make sure the API uses `Indexer.Name` when it's returned via the API.

Not sure how that'd affect blocklist/queued releases, but it'd save extra look ups for each grab.

Reply:
Are we storing RemoteEpisode in the DB anywhere? PendingReleases or anything?

Reply:
With the other refactor of logic down to one call this really didn't buy us much given we still need to call and get the definition when a release is pushed.